src/azure-cli/azure/cli/command_modules/acr/_validators.py

```diff
@@ -103,32 +103,37 @@ def validate_retention_days(namespace):
         raise CLIError("Invalid value for days: should be from 0 to 365")
 
 
-def validate_registry_name(cmd, namespace):
+def validate_registry_name(cli_ctx, registry_name):
     """Omit login server endpoint suffix."""
-    registry = namespace.registry_name
-    if registry is None:
+    if registry_name is None:
         return
-    suffixes = cmd.cli_ctx.cloud.suffixes
+    suffixes = cli_ctx.cloud.suffixes
+    dnl_hash = registry_name.find("-")
+
+    if dnl_hash > 0:
+        logger.warning(
+            "Registry name is %s. The following suffix '%s' is automatically omitted.",
+            registry_name[:dnl_hash],
+            registry_name[dnl_hash:])
+        registry_name = registry_name[:dnl_hash]
+        dnl_suffix = registry_name[dnl_hash:]
```

Comment:
suffix is already stripped in the previous line. Anyway we can remove this code. We do not have to return the dnl_suffix. We should not be constructing login server